File: api/Models/Sponsorship.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace api.Models
{
    public class Sponsorship
    {
        public int Id {get; set;}
        public string Sponsoree {get; set;}
        public string Event {get; set;}
        public string EventStartDate {get; set;}
        public string EventEndDate {get; set;}
        public string ContractPrice {get; set;}
    }
}

File: dashboard/src/sponsorship/sponsorship.js
import {inject} from 'aurelia-framework';
import {EventAggregator} from 'aurelia-event-aggregator';
import {SponsorshipCreated,SponsorshipViewed, ToggleSponsorshipFormVisibility} from '../messages';
import {WebAPI} from '../web-api';

@inject(WebAPI, EventAggregator)
export class Sponsorship {
    sponsorship = {};
    constructor(api, ea){
        this.api = api;
        this.ea = ea;
    }

    get canSave() {       
        return this.sponsorship.sponsoree && this.sponsorship.event && this.sponsorship.eventStartDate && this.sponsorship.eventEndDate && this.sponsorship.contractPrice && this.sponsorship.contractPrice >= 100;
    }


    save() {
        this.api.saveSponsorship(this.sponsorship).then(sponsorship => {
            this.sponsorship = sponsorship;
            this.ea.publish(new SponsorshipCreated(this.sponsorship));
            this.sponsorship = {};
        });
    }
}